ticker.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaeger
+
+import (
+	"math/rand"
+	"time"
+)
+
+var random *rand.Rand
+
+func init() {
+	random = rand.New(rand.NewSource(time.Now().UnixNano()))
+}
+
+// Ticker  is a customer Ticker, it perform like a normal time.Ticker when operation is success,
+// When Operation is fail, it will use exponential backoff with a jitter.
+type Ticker struct {
+	C        <-chan time.Time
+	timer    *time.Timer
+	duration time.Duration
+	failCnt  uint
+}
+
+// NewTicker construct a new Ticker
+func NewTicker(duratin time.Duration) *Ticker {
+	t := &Ticker{
+		timer:    time.NewTimer(duratin),
+		duration: duratin,
+	}
+	t.C = t.timer.C
+	return t
+}
+
+// Stop turns off a Ticker. After Stop, no more ticks will be sent.
+func (t *Ticker) Stop() {
+	t.timer.Stop()
+}
+
+// UpdateTimer update the timer use operation result, must be called after every operation
+func (t *Ticker) UpdateTimer(err error) {
+	d := t.next(err)
+	t.timer.Reset(d)
```

Comment:
Documentation about [Reset](https://golang.org/pkg/time/#Timer.Reset) seems really specific about usage.

> If a program has already received a value from t.C, the timer is known to have expired and the channel drained, so t.Reset can be used directly. If a program has not yet received a value from t.C, however, the timer must be stopped

The way you have structured this code I think it will always be true that a value has been received on `t.C`, but we should add a comment that `UpdateTimer` must be called after receiving from the channel.

Honestly that first sentence scares me. Is there a race condition here where the channel gets a message added between draining the channel and `Reset`?

Reply:
That's right. Since the value had been received before I called the Reset, I think it's safe for this situation.

I think it's more safe to stop the timer first and then try to drain the t.C if not empty, finnaly call the Reset as the Documentation mentioned . I will fix this.